src/components/app.js

```diff
@@ -78,13 +88,19 @@ const App = () => {
       });
   };
   const handleChangePlan = (p) => {
+    if (p === "custom") {
+      setJSONText(localStorage.getItem("customJSON"))
+    }
     setPlan(p);
   };
   const handleOpenGenerate = () => {
     setModal(true);
   };
   const generateCustomTable = () => {
+    localStorage.setItem("customJSON", jsonText);
+    localStorage.setItem("plan", "custom")
```

Comment:
The logic almost same as line 55-65, I think it should be better to handle it in that part.